.github/workflows/main.yml

```diff
@@ -1,21 +1,71 @@
-name: CI on Windows
+name: CI
 
 on: [push, pull_request]
 
 jobs:
-  build:
+  ubuntu:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        name: ["py35", "py36", "py37", "py38", "nightly"]
+        include:
+          - name: "py35"
+            python: "3.5"
+            tox_env: "du12"
+          - name: "py36"
+            python: "3.6"
+            tox_env: "du13"
+          - name: "py37"
+            python: "3.7"
+            tox_env: "du14"
+          - name: "py38"
+            python: "3.8"
+            tox_env: "du15"
+            use_coverage: true
+          - name: "nightly"
+            python: "3.10-dev"
+            tox_env: "du16"
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        if: "!endsWith(matrix.python, '-dev')"
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Set up Python ${{ matrix.python }} (deadsnakes)
+        uses: deadsnakes/action@v1.0.0
+        if: endsWith(matrix.python, '-dev')
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Check Python version
+        run: python --version
+      - name: Install graphviz
+        run: sudo apt-get install graphviz
+      - name: Install dependencies
+        run: pip install -U tox codecov
+      - name: Run tox
+        run: tox -- -vv -e ${{ matrix.tox_env }}
+      - name: codecov
+        if: matrix.use_coverage
+        env:
+          PYTEST_ADDOPTS: "--cov ./ --cov-append --cov-config setup.cfg"
+        run: codecov -e ${{ matrix.tox_env }}
+
+  windows:
     runs-on: windows-latest
     strategy:
       matrix:
         architecture: [x86, x64]
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python
-      uses: actions/setup-python@v1
-      with:
-        architecture: ${{ matrix.architecture }}
-    - name: Install dependencies
-      run:  pip install -U tox
-    - name: Run Tox
-      run:  tox -e py -- -vv
+      - uses: actions/checkout@v2
```

Comment:
@tk0miya Which style would you prefer for an array?

```yml
# 1. no indent
steps:
- ...

# 2. indent
steps:
  - ...
```

Reply:
No opinion. Either is good to me.

Reply:
ok I'll use the current style.